bayes_opt/domain_reduction.py

```diff
@@ -180,20 +212,47 @@ def _trim(self, new_bounds: np.array, global_bounds: np.array) -> np.array:
 
         return new_bounds
 
-    def _window_bounds_compatibility(self, global_bounds: np.array) -> bool:
-        """Checks if global bounds are compatible with the minimum window sizes.
+    def _window_bounds_compatibility(self, global_bounds: np.ndarray):
+        """Check if global bounds are compatible with the minimum window sizes.
+        
+        Parameters
+        ----------
+        global_bounds : np.ndarray
+            The maximum allowable bounds for each parameter.
+
+        Raises
+        ------
+        ValueError
+            If global bounds are not compatible with the minimum window size.
         """
         for i, entry in enumerate(global_bounds):
             global_window_width = abs(entry[1] - entry[0])
             if global_window_width < self.minimum_window[i]:
                 raise ValueError(
                     "Global bounds are not compatible with the minimum window size.")
 
-    def _create_bounds(self, parameters: dict, bounds: np.array) -> dict:
+    def _create_bounds(self, parameters: dict, bounds: np.ndarray) -> dict:
+        """Create a dictionary of bounds for each parameter.
+        
+        Parameters
+        ----------
+        parameters : dict
+            The parameters for which to create the bounds.
+        """
         return {param: bounds[i, :] for i, param in enumerate(parameters)}
 
     def transform(self, target_space: TargetSpace) -> dict:
-        """Reduces the bounds of the target space.
+        """Transform the bounds of the target space.
+        
+        Parameters
+        ----------
+        target_space : TargetSpace
+            TargetSpace this DomainTransformer operates on.
+        
+        Returns
+        -------
+        dict
```

Comment:
```suggestion
        bounds : dict
```

Reply:
In general, should we have the fields under "Returns" have the following format? object_name : object_type ?

I understand that just reporting the type suffices, but I think a name could be more clear. Thoughts?

Reply:
I'm fine with either -- imo it _should_ be clear from the function name what it returns ideally, but it doesn't add much overhead.

Reply:
my vote: it's only necessary to specify return type, not return name since the name of the returned object depends on how the function is called.